skimage/measure/_regionprops.py

```diff
@@ -941,19 +941,28 @@ def regionprops_table(
     extra_properties=None,
     spacing=None,
 ):
-    """Compute image properties and return them as a pandas-compatible table.
+    """Compute region properties and return them as a pandas-compatible table.
 
-    The table is a dictionary mapping column names to value arrays. See Notes
-    section below for details.
+    The return value is a dictionary mapping property names to value arrays.
+    This dictionary can be used as input to ``pandas.DataFrame`` to result in
+    a "tidy" [1]_ table with one region per row and one property per column.
+
+    Use this function typically when you want to do downstream data analysis,
+    or save region data to disk in a structured way. One downside of this
+    function is that it breaks multi-dimensional properties into independent
+    columns; for example, the region centroids of a 3D image end up in three
+    different columns, one per dimension. If you need to do complex
+    computations with the region properties, using
+    :func:`skimage.measure.regionprops` might be more fitting.
```

Comment:
This could also go into the "Notes" section but since it's very useful context I'm totally fine with keeping it here. :) Nice improvement!